### PR TITLE
test: setup new integration lnd service tests

### PR DESCRIPTION
**Pull request by vindard**
## Description

Add new lnd-service integration tests and move tests from legacy into here

### CODE REVIEW

File: test/helpers/lightning.ts [hunk 1]
```diff
@@ -188,6 +188,65 @@ export const openChannelTesting = async ({
...
+  baseLogger.debug("mining blocks and waiting for channel being opened")
+  await Promise.all([
+    // error: https://github.com/alexbosworth/ln-service/issues/122
+    // once(sub, 'channel_opened'),
+    waitFor(() => lndNewChannel && lndPartnerNewChannel),
+    mineBlockAndSync({ lnds: [lnd, lndPartner] }),
+  ])
+
+  if (lndPartner === lnd1) {
+    expect(lndPartnerNewChannel.is_partner_initiated).toBe(true)
```
Comment (dolcalmi) [inline test/helpers/lightning.ts:239]: this is a bad practice... we should not have validation in helper methods

File: test/integration/services/lnd-service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    const paid = await lndService.payInvoiceViaPaymentDetails({
+      decodedInvoice: lnInvoice,
+      btcPaymentAmount,
+      maxFeeAmount: undefined,
+    })
+    if (paid instanceof Error) throw paid
+
+    // Confirm payment exists in lnd
+    const retrievedPayment = await lndService.lookupPayment({ paymentHash })
+    expect(retrievedPayment).not.toBeInstanceOf(Error)
```
Comment (dolcalmi) [inline test/integration/services/lnd-service.spec.ts:329]: Please validates against specific types (applies to all the error expects)

File: test/legacy-integration/02-user-wallet/02-send-lightning.spec.ts.skip
Comment (dolcalmi) [inline test/legacy-integration/02-user-wallet/02-send-lightning.spec.ts.skip:1]: are you sure this is completely covered? or why is skipped?

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: test/integration/services/lnd-service.spec.ts (hunk 2 above)
Comment (vindard) [reply test/integration/services/lnd-service.spec.ts:329]: I usually validate against specific error types for positive validation (expect a specific error), but keep it generic when I don't want any error at all coming back from a call

File: test/legacy-integration/02-user-wallet/02-send-lightning.spec.ts.skip
Comment (vindard) [reply test/legacy-integration/02-user-wallet/02-send-lightning.spec.ts.skip:1]: It's mostly covered and across the current work will be fully covered in the next day or 2. Not sure it's worth the investment trying to keep these working now given how aggressively we're removing tests from that file and how interconnected the tests are in there.

File: test/helpers/lightning.ts (hunk 1 above)
Comment (vindard) [reply test/helpers/lightning.ts]: Good catch, I'd just copy-pasted the function just above and removed the accounting lines. Did a bit more direct cleanup of that function here: https://github.com/GaloyMoney/galoy/pull/3054/commits/b5aa657b4cf2b2ba5ee813d8fedac0c68ebbbf51

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**